unit_tests/test_version_utils.py

```diff
@@ -388,3 +389,23 @@ def test_scylla_version_grouped_regexp(full_version, version, date, commit_id):
     assert parsed_version.group("version") == version
     assert parsed_version.group("date") == date
     assert parsed_version.group("commit_id") == commit_id
+
+
+@pytest.mark.integration
+@pytest.mark.need_network
+@pytest.mark.skip(reason="those are integration tests only")
+@pytest.mark.parametrize("version, expected", (
+    ("4.6.rc2-20220102.e8a1cfb6f", "scylladb/scylla:4.6.rc2"),
+    ("5.0.1-0.20220719.b177dacd3", "scylladb/scylla:5.0.1"),
+    ("5.2.0~dev-0.20220824.6ce5e9079c1e", "scylladb/scylla-nightly:5.2.0-dev-0.20220824.6ce5e9079c1e"),
+    ("5.1.0~dev-0.20220726.29c28dcb0c33", "scylladb/scylla-nightly:5.1.0-dev-0.20220726.29c28dcb0c33"),
+    ("2022.2.dev-20220515.no_such_sha", "scylladb/scylla-enterprise:latest"),
+    ("5.2.0~dev-20220515.no_such_sha", "scylladb/scylla:latest"),
+))
+def test_get_docker_image_by_version(version, expected):
+    assert get_docker_image_by_version(scylla_version=version) == expected
+
+
+def test_get_docker_image_by_version_broken_string_version():
+    with pytest.raises(AssertionError):
+        get_docker_image_by_version(scylla_version="broken_version")
```

Comment:
you mean this should fail for short versions ?

Reply:
I'm not sure what exactly do you mean, it meant for finding images based on the version `scylla --version` supply, it can't supply short versions, and we care for exact ones...

Reply:
I meant adding unit test coverage with the expected behavior.

Reply:
Added:
```
    ("5.1", "scylladb/scylla:5.1"),
    ("5.1.0~rc1", "scylladb/scylla:5.1.0-rc1"),
    ("5.0.1", "scylladb/scylla:5.0.1"),
```

keep in mind this function doesn't force the input to be semver compliant, it's just to be able to match the docker image with the scylla under test. worst case it would take latest released version.